prow/cmd/checkconfig/main.go

```diff
@@ -375,3 +385,19 @@ func validateDecoratedJobs(configAgent config.Agent) error {
 	}
 	return nil
 }
+
+func validateNeedsOkToTestLabel(configAgent config.Agent) error {
+	var queryErrors []error
+	for i, query := range configAgent.Config().Tide.Queries {
+		for _, label := range query.Labels {
+			if label == lgtm.LGTMLabel {
+				for _, label := range query.MissingLabels {
+					if label == trigger.NeedsOkToTest {
+						queryErrors = append(queryErrors, fmt.Errorf("the tide query at position %d forbids the %q label, which is not recommended", i, trigger.NeedsOkToTest))
```

Comment:
The error message should indicate that the problem is not that the `needs-rebase` label is forbidden, but rather that the `needs-rebase` label is forbidden *and* the `lgtm` label is required *and* managed by the `lgtm` plugin.
This reason for this warning isn't immediately obvious like other types of warnings so we may want to link to https://github.com/kubernetes/test-infra/blob/master/prow/cmd/tide/maintainers.md#best-practices as an explanation.